takwimu/views.py

```diff
@@ -349,7 +349,7 @@ def get(self, request, *args, **kwargs):
             # eg. './icon.png' from 'style.css' => 'style.css/icon.png'
             path_parts = []
             for index, path_part in enumerate(path.split('/')):
-                if '.' not in path_part or index == len(path_parts) - 1:
+                if '.' not in path_part or index == len(path_parts):
```

Comment:
I think this is pretty much the same @KhadijaMahanga but with one flaw: `index` will _never_ equal `len(path_parts)`. From what we discussed (and if I remember correctly), we need to check if `path_part` is the last or the only part in `path_parts` first. Only when it is **not** true, then and only then check if it could be a file. i.e.

```python
...
            for index, path_part in enumerate(path.split('/')):
                if index == len(path_parts) - 1 or '.' not in path_part:
...

Reply:
Yeah the above makes sense ^